daphne/src/testing.rs

```diff
@@ -470,32 +474,134 @@ impl DapLeader<BearerToken> for MockAggregator {
         Ok(reports)
     }
 
-    async fn init_collect_job(&self, _collect_req: &CollectReq) -> Result<Url, DapError> {
-        unreachable!("not implemented");
+    // Called after receiving a CollectReq from Collector.
+    async fn init_collect_job(&self, collect_req: &CollectReq) -> Result<Url, DapError> {
+        let mut rng = thread_rng();
+        let task_config = self
+            .get_task_config_for(&collect_req.task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
+
+        let mut leader_state_store_mutex_guard = self
+            .leader_state_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+        let leader_state_store = leader_state_store_mutex_guard.deref_mut();
+
+        // Check whether a collect job already exists for this task_id.
+        // If so, push CollectReq to LeaderState and retrieve the pre-computed collect URI.
+        // If not, create new LeaderState with new collect URI and CollectReq.
+        if leader_state_store.contains_key(&collect_req.task_id) {
+            let state = leader_state_store.get_mut(&collect_req.task_id).unwrap();
```

Comment:
Alternatively:
```rust
if lest Some(state) = leader_state_store.contains_key(&collect_Req.task_id) {
    // use state
} else {
   // create state
}
```

Reply:
Using this would've been nice, but there are restrictions in the code that would require me to write the same code twice, which I would like to avoid.